spec/controllers/reviews_controller_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'spec_helper'
+
+describe ReviewsController do
+  context "when review is created with valid input" do
+    let(:review) { Fabricate.attributes_for(:review) }
+    let(:video) { Fabricate(:video) }
+    before do
+      post :create, video_id: video.id, review: review
+    end
+    describe "POST /videos/:video_id/reviews" do
+      it "creates a new @review" do
+        expect(Review.count).to eq(1)
+      end
+      it "redirects to the @video video_path" do
+        expect(response).to redirect_to video_path(video)
+      end
+      it "sets the notice" do
+        expect(flash[:notice]).not_to be_blank
+      end
+    end
+  end
+
+  context "when review being is updated" do
+    describe "GET /videos/:video_id/reviews/:id/edit" do
+      let(:video) { Fabricate(:video) }
+      let(:review) { Fabricate(:review, video_id: video.id) }
+      before { get :edit, video_id: video.id, id: review }
+      it "sets the @review object" do
+        expect(assigns(:review)).to eq(review)
+      end
+    end
+
+    describe "PUT /videos/:video_id/reviews/:id" do
+      let(:video) { Fabricate(:video) }
+      let(:review) { Fabricate(:review, video_id: video.id) }
+      let(:new_attributes) do 
+        { rating: 2, review: "New review." }
+      end
+      before do
+        put :update, video_id: video.id, id: review
+        review.reload
+      end
+      it "should redirect to video path" do
+        expect(response).to redirect_to video_path(video)
+      end
+      it "updates the attributes of the review object" do
+        expect(assigns(:review)).to eq(review)
+      end
+      it "sets the flash notice" do
+        expect(flash[:notice]).not_to be_blank
+      end
+    end
+  end
+
+  context "when invalid input" do
+    let(:video) { Fabricate(:video) }
+    before do
+      post :create, video_id: video.id, review: { review: "This is a post witout a rating.", video_id: video.id , user_id: Fabricate(:user).id }
```

Comment:
You could also Fabricate a review with only the `review` attribute. Then you can do `Fabricate(:review, video_id: video.id, user_id: Fabricate(:user).id)`. You could also use the virtual attributes given you by the associations: `Fabricate(:review, video: video, user: Fabricate(:user))`